keydata.go

```diff
@@ -310,8 +299,13 @@ func (d *KeyData) snapModelAuthKey(auxKey AuxiliaryKey) ([]byte, error) {
 }
 
 // ID returns the unique ID for this key data.
-func (d *KeyData) ID() KeyID {
-	return d.id
+func (d *KeyData) ID() (KeyID, error) {
```

Comment:
This is fine in itself. Fwiw  in snapd we tend to call this kind of methods that return a unique/key-mapping identifier Unique.

But my comment https://github.com/snapcore/secboot/pull/134/files#r613927267 still needs to be addressed somehow.